drivers/bluetooth/hci/hci_stm32wba.c

```diff
@@ -0,0 +1,380 @@
+		buf = treat_evt(data, len);
+		break;
+	case HCI_ACL:
+		buf = treat_acl(data, len + 1, ext_data, ext_len);
+		break;
+	case HCI_ISO:
+	case HCI_SCO:
+		buf = treat_iso(data, len + 1, ext_data, ext_len);
+		break;
+	default:
+		buf = NULL;
+		LOG_ERR("Unknown HCI type %u", pkt_indicator);
+	}
+
+	if (buf) {
+		bt_recv(buf);
+	} else {
+		err = -ENOMEM;
+		ll_state_busy = 1;
+	}
+
+	return err;
+}
+
+uint8_t BLECB_Indication(const uint8_t *data, uint16_t length,
+			 const uint8_t *ext_data, uint16_t ext_length)
+{
+	int ret = 0;
+	int err;
+
+	LOG_DBG("length: %d", length);
+	if (ext_length != 0) {
+		LOG_DBG("ext_length: %d", ext_length);
+	}
+
+	k_sem_take(&hci_sem, K_FOREVER);
+
+	err = receive_data((uint8_t *)data, (size_t)length - 1,
+			(uint8_t *)ext_data, (size_t)ext_length);
+
+	k_sem_give(&hci_sem);
+
+	HostStack_Process();
+
+	if (err) {
+		ret = 1;
+	}
+
+	return ret;
+}
+
+static int bt_hci_stm32wba_send(struct net_buf *buf)
+{
+	uint16_t event_length;
+	uint8_t pkt_indicator;
+	uint8_t tx_buffer[300];
+
+	k_sem_take(&hci_sem, K_FOREVER);
+
+	LOG_DBG("buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
+
+	switch (bt_buf_get_type(buf)) {
+	case BT_BUF_ACL_OUT:
+		pkt_indicator = HCI_ACL;
+		break;
+	case BT_BUF_CMD:
+		pkt_indicator = HCI_CMD;
+		break;
+	case BT_BUF_ISO_OUT:
+		pkt_indicator = HCI_ISO;
+		break;
+	default:
+		LOG_ERR("Unknown type %u", bt_buf_get_type(buf));
+		k_sem_give(&hci_sem);
+		return -EIO;
+	}
+	net_buf_push_u8(buf, pkt_indicator);
+
+	memcpy(&tx_buffer, buf->data, buf->len);
+
+	event_length = BleStack_Request(tx_buffer);
+	LOG_DBG("event_length: %u", event_length);
+
+	if (event_length) {
+		receive_data((uint8_t *)&tx_buffer, (size_t)event_length, NULL, 0);
+	}
+
+	k_sem_give(&hci_sem);
+
+	net_buf_unref(buf);
+
+	return 0;
+}
+
+static tBleStatus bt_ble_ctlr_init(void)
+{
+	BleStack_init_t init_params_p = {0};
+	tBleStatus err;
+
+	init_params_p.numAttrRecord           = CFG_BLE_NUM_GATT_ATTRIBUTES;
+	init_params_p.numAttrServ             = CFG_BLE_NUM_GATT_SERVICES;
+	init_params_p.attrValueArrSize        = CFG_BLE_ATT_VALUE_ARRAY_SIZE;
+	init_params_p.prWriteListSize         = CFG_BLE_ATTR_PREPARE_WRITE_VALUE_SIZE;
+	init_params_p.attMtu                  = CFG_BLE_ATT_MTU_MAX;
+	init_params_p.max_coc_nbr             = CFG_BLE_COC_NBR_MAX;
+	init_params_p.max_coc_mps             = CFG_BLE_COC_MPS_MAX;
+	init_params_p.max_coc_initiator_nbr   = CFG_BLE_COC_INITIATOR_NBR_MAX;
+	init_params_p.numOfLinks              = CFG_BLE_NUM_LINK;
+	init_params_p.mblockCount             = CFG_BLE_MBLOCK_COUNT;
+	init_params_p.bleStartRamAddress      = (uint8_t *)buffer;
+	init_params_p.total_buffer_size       = BLE_DYN_ALLOC_SIZE;
+	init_params_p.bleStartRamAddress_GATT = NULL;
+	init_params_p.total_buffer_size_GATT  = 0;
+	init_params_p.options                 = CFG_BLE_OPTIONS;
+	init_params_p.debug                   = 0U;
+
+	err = BleStack_Init(&init_params_p);
+	if (err < 0) {
+		return -EIO;
```

Comment:
Is `-EIO` a valid value for `tBleStatus`? Where is this type and its possible values defined?

Reply:
Good catch. Those are defined in `lib/stm32wba/hci/ble_defs.h` from https://github.com/zephyrproject-rtos/hal_stm32/pull/187. Now testing against BLE_STATUS_SUCCESS.

Reply:
> Good catch. Those are defined in `lib/stm32wba/hci/ble_defs.h` from [zephyrproject-rtos/hal_stm32#187](https://github.com/zephyrproject-rtos/hal_stm32/pull/187). Now testing against BLE_STATUS_SUCCESS.

Sure, but shouldn't the return value of this `bt_ble_ctrl_init()` function be `int` instead of `tBleStatus` since you're returning `-EIO`?

Reply:
Sure, indeed. Fixing